compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs

```diff
@@ -1381,7 +1381,8 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
         trait_pred: ty::PolyTraitPredicate<'tcx>,
     ) -> bool {
         let hir = self.tcx.hir();
-        let parent_node = hir.get_parent_node(obligation.cause.body_id);
+        let body_hir = hir.local_def_id_to_hir_id(obligation.cause.body_id);
+        let parent_node = hir.get_parent_node(body_hir);
         let node = hir.find(parent_node);
```

Comment:
The new `body_id: LocalDefId` is typically the parent of the old `body_id: BodyId`, as its `body_owner_def_id`.
Therefore, you should not look at `get_parent_node`, and directly inspect the node at `body_id`.
```suggestion
        let node = hir.find_by_def_id(obligation.cause.body_id);
```

Reply:
Oh, thanks